lib/widgets/picker_fields/card_settings_file_picker.dart

```diff
@@ -0,0 +1,192 @@
+    return ((bytes / pow(1024, i)).toStringAsFixed(decimals)) +
+        ' ' +
+        suffixes[i];
+  }
+}
+
+class _CardSettingsFilePickerState extends FormFieldState<Uint8List> {
+  @override
+  CardSettingsFilePicker get widget => super.widget as CardSettingsFilePicker;
+
+  void _showDialog() {
+    FilePickerCross filePicker = FilePickerCross(
+      fileExtension: widget.fileExtension,
+      type: widget.fileType,
+    );
+    filePicker.pick().then((value) => setState(() {
+          final _file = filePicker.toUint8List();
+          didChange(_file);
+          if (widget.onChanged != null) widget.onChanged(_file);
+        }));
+  }
+
+  Widget _build(BuildContext context) {
+    String formattedValue = (value == null)
+        ? ''
+        : CardSettingsFilePicker.formatBytes(value.length, 2);
+
+    if (showCupertino(context, widget.showMaterialonIOS))
+      return cupertinoSettingsDatePicker(formattedValue);
+    else
+      return materialSettingsDatePicker(formattedValue);
+  }
+
+  void onTap() {
+    if (value == null) {
+      _showDialog();
+    } else {
+      showPlatformDialog<void>(
+        context: context,
+        builder: (context) => PlatformAlertDialog(
+          title: Text(
+            widget.unattachDialogTitle,
+            style: isMaterial(context)
+                ? Theme.of(context).textTheme.headline6.copyWith(
+                    color: Theme.of(context).textTheme.headline1.color)
+                : null,
+          ),
+          actions: [
+            PlatformDialogAction(
+              child: PlatformText(widget.unattachDialogCancel),
+              onPressed: () => Navigator.of(context).pop(),
+            ),
+            PlatformDialogAction(
+              child: PlatformText(widget.unattachDialogConfirm),
+              cupertino: (_, __) =>
+                  CupertinoDialogActionData(isDestructiveAction: true),
+              onPressed: () {
+                didChange(null);
+                if (widget.onChanged != null) widget.onChanged(null);
+                Navigator.of(context).pop();
+              },
+            )
+          ],
+        ),
+      );
+    }
+  }
+
+  Widget cupertinoSettingsDatePicker(String formattedValue) {
+    return Container(
+      child: widget?.visible == false
+          ? null
+          : GestureDetector(
+              onTap: onTap,
+              child: CSControl(
+                nameWidget: widget?.requiredIndicator != null
+                    ? Text((widget?.label ?? "") + ' *')
+                    : Text(widget?.label),
+                contentWidget: Text(
+                  formattedValue,
+                  style: widget?.style ?? Theme.of(context).textTheme.subtitle1,
+                  textAlign: widget?.contentAlign ??
+                      CardSettings.of(context).contentAlign,
+                ),
+                style: CSWidgetStyle(icon: widget?.icon),
+              ),
+            ),
+    );
```

Comment:
Your code still refers to DatePicker in some of the names, please fix.

Reply:
Does this work with web?

Reply:
> Does this work with web?

Yep, check file_picker_cross docs.

Reply:
FYI I renamed the methods